src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -390,3 +378,19 @@ def _finalize_pnad_side_predicates(pnad: PartialNSRTAndDatastore) -> None:
             for atom in next_atoms - segment.final_atoms:
                 side_predicates.add(atom.predicate)
         pnad.op = pnad.op.copy_with(side_predicates=side_predicates)
+
+    def _assert_all_data_in_exactly_one_datastore(
+            self, pnads: List[PartialNSRTAndDatastore]) -> None:
+        """Assert that every demo datapoint appears in exactly one datastore
```

Comment:
maybe i'm being dum but i don't see how this checks for EXACTLY one?

Reply:
the `len(x) == len(set(x))` check

Reply:
yup was being dumb cool